From cbb1416dd3138781e83105307d4c2a9262caaa27 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 14 Nov 2024 19:53:43 +0800
Subject: [PATCH 37/68] grub.d: 00-header: disable graphical output for
 MacBook6,2

MacBook Pro 15'' (Mid-2010, MacBookPro6,2) are known to have broken
graphical terminal output. Use console output for this model.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 util/grub.d/00_header.in | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/util/grub.d/00_header.in b/util/grub.d/00_header.in
index 7c1b73a7f..acfba467a 100644
--- a/util/grub.d/00_header.in
+++ b/util/grub.d/00_header.in
@@ -254,6 +254,17 @@ EOF
   ;;
 esac
 
+cat << EOF
+# Hardware quirks:
+#
+# MacBook Pro 15'' (Mid-2010, MacBookPro6,2) are known to have broken
+# graphical terminal output. Use console output for this model.
+smbios --type 1 --get-string 5 --set system_model
+if [ "x$system_model" = "xMacBookPro6,2" ]; then
+	terminal_output console
+fi
+EOF
+
 if [ "x$gfxterm" = x1 ]; then
     if [ "x$GRUB_THEME" != x ] && [ -f "$GRUB_THEME" ] \
 	&& is_path_readable_by_grub "$GRUB_THEME"; then
-- 
2.51.1

